Modules/timemodule.c

```diff
@@ -2051,34 +2051,51 @@ PyInit_time(void)
 static int
 pysleep(_PyTime_t secs)
 {
-    _PyTime_t deadline, monotonic;
 #ifndef MS_WINDOWS
 #ifdef HAVE_CLOCK_NANOSLEEP
     struct timespec timeout_abs;
 #else
     struct timeval timeout;
 #endif
+    _PyTime_t deadline, monotonic;
     int err = 0;
     int ret = 0;
-#else
-    _PyTime_t millisecs;
-    unsigned long ul_millis;
-    DWORD rc;
-    HANDLE hInterruptEvent;
-#endif
 
     if (get_monotonic(&monotonic) < 0) {
         return -1;
     }
     deadline = monotonic + secs;
-#if defined(HAVE_CLOCK_NANOSLEEP) && !defined(MS_WINDOWS)
+#ifdef HAVE_CLOCK_NANOSLEEP
     if (_PyTime_AsTimespec(deadline, &timeout_abs) < 0) {
         return -1;
     }
 #endif
+#else
+    _PyTime_t millisecs;
+    DWORD ul_millis;
+    DWORD rc;
+    FILETIME FileTime;
+    LARGE_INTEGER deadline;
+    HANDLE hTimer = NULL;
+
+    GetSystemTimePreciseAsFileTime(&FileTime);
+    CopyMemory(&deadline, &FileTime, sizeof(FILETIME));
+
+    /* convert to 100 nsec unit */
+    deadline.QuadPart += (_PyTime_AsMicroseconds(secs, _PyTime_ROUND_CEILING) * 10);
```

Comment:
This operation loses precision. You can convert to nanoseconds and then divide by 100, no?

Reply:
`_PyTime_AsNanoseconds()` is not yet implemented, so need to create it firstly. However, Windows is still weak in sleep (in this implementation too), which means, even though if I implement it, the accuracy of sleeping will be between milisecs and microsecs because it is too far from a real-time OS. My opinion is that, it useless to take more effort in it.

Reply:
I wrote PR #28350 to add _PyTime_AsNanoseconds()... and make a lot of refactoring changes.

Reply:
I also added _PyTime_As100Nanoseconds() to be able to calculate lossless 100 nsec unit.

Reply:
I don't see the point. Just _PyTime_AsNanoseconds(), no?

Reply:
This is my  _PyTime_AsNanoseconds() [commit](https://github.com/Livius90/cpython/commit/ac8ab07a473c72a3532c887fdc1a2e09a6194ce2). I saw the new _PyTime_AsNanoseconds() commti and merged well to my working branch, but i need to do divide by 100 with ceiling, so it is better to make a special function for it as the others unit convertions.